### PR TITLE
PLAT-13089 CLI Amplitude

### PR DESCRIPTION
**Pull request by robester0403**
## Ticket

PLAT-13089
_Related tickets:_
_Related PRs:_

## Type of PR

- [ ] Bugfix
- [x] New feature
- [ ] Minor changes

Did you make changes to modules or created a new module?

- [ ] Yes, and have read the [modules updates checklist](https://github.com/crowdbotics/modules/#modules-updates-checklist) documentation and followed the instructions.
- [x] No.

## Changes introduced

Amplitude sending data for all commands used. Opt In and Opt out

## Test and review

_Describe how a reviewer should test your PR_

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -72,6 +74,25 @@ function dispatcher() {
...
+    const eventProperties = {
+      full_command: process.argv.slice(2).join(" "), // all of the commands in the user input
+      action: command // Just the first command
+    };
+
+    // track the event
+    track("Crowdbotics CLI", eventProperties, {
+      user_id: username
+    });
+  }
```
Comment (alaisgomes) [inline index.js:94]: Could this maybe all be in a separate function? 

And if possible, all inside a try/catch. Maybe do nothing if it error out for now...  If something is messed up with Amplitude, we don't want the user to be blocked in their actions. Later on we can track errors somehow.

File: scripts/amplitude/config.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { configFile } from "../utils/configFile.js";
+import { HOST_CONFIG_NAME } from "../utils/constants.js";
+
+export const HAS_ASKED_OPT_IN_NAME = "has-asked-opt-in";
+export const OPT_IN_NAME = "opt-in";
+
+export const AMPLITUDE_API_KEY = configFile.get(HOST_CONFIG_NAME) === "https://app.crowdbotics.com/" ? "production-key-placeholder-text" : "development-key-placeholder-text";
```
Comment (alaisgomes) [inline scripts/amplitude/config.js:7]: Question: Where will the key be coming from?

Comment (alaisgomes) [reply scripts/amplitude/config.js:7]: I was thinking about this, we will probably need to store the production key in the constants file. I don't  love this, but not sure if we can use environment variables, but we could test.

**Review by alaisgomes — COMMENTED** [review]
This looks good! Minor request for code breakdown and starting a "thread" on API tokens

File: index.js (hunk 1 above)
Comment (robester0403) [reply index.js]: I put an extra property to capture the error through another tracking event... though it is questionable if the try fails then can the catch actually send an event to amplitude properly.

Probably need to rethink this.

File: scripts/amplitude/config.js (hunk 2 above)
Comment (robester0403) [reply scripts/amplitude/config.js:7]: Have been looking into this for a bit so didn't respond yet. We basically have to give the user the API key at some point cause they are going to install our package on their machine.

Trying to get a decent solution for this but ultimately it depends on how far we want to go to hide the api keys.

Just noticing that other well used integrations like Zoom, github, etc passes the responsibility to the user. Ie they require you to give you some sort of key to use their services.

Comment (alaisgomes) [reply scripts/amplitude/config.js:7]: I am not super worried about amplitude API keys, but I could be in the wrong. I would be ok to make it the default value as a constant, like the default API url is the production url. As far as tracking data usage

File: index.js (hunk 1 above)
Comment (alaisgomes) [reply index.js]: It might not be a bad idea, maybe if we group it as a different event like `Crowdbotics CLI Errors`. But we could check with product too

**Review by alaisgomes — APPROVED** [review]

**Pull request merged**